pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -64,58 +67,81 @@ class Simulation:
                 "The provided sequence has to be a valid "
-                "pulser.Sequence instance."
+                "SequenceSamples instance."
             )
-        if sequence.is_parametrized() or sequence.is_register_mappable():
+        if sampled_seq.max_duration == 0:
+            raise ValueError("SequenceSamples is empty.")
+        # Check compatibility of register and device
+        self.__device = device
+        self.__device.validate_register(register)
+        self.__register = register
+        # Check compatibility of samples and device:
+        if (
+            sampled_seq._slm_mask.end > 0
+            and not self.__device.supports_slm_mask
+        ):
             raise ValueError(
-                "The provided sequence needs to be built to be simulated. Call"
-                " `Sequence.build()` with the necessary parameters."
+                "Samples use SLM mask but device does not have one."
             )
-        if not sequence._schedule:
-            raise ValueError("The provided sequence has no declared channels.")
-        if all(
-            sequence._schedule[x][-1].tf == 0
-            for x in sequence.declared_channels
-        ):
+        if not sampled_seq.used_bases <= device.supported_bases:
             raise ValueError(
-                "No instructions given for the channels in the sequence."
+                "Bases used in samples should be supported by device."
             )
-        self._seq = sequence
-        self._interaction = "XY" if self._seq._in_xy else "ising"
-        self._qdict = self._seq.qubit_info
-        self._size = len(self._qdict)
-        self._modulated = with_modulation
-        if self._modulated and sequence._slm_mask_targets:
-            raise NotImplementedError(
-                "Simulation of sequences combining an SLM mask and output "
-                "modulation is not supported."
+        # Check compatibility of masked samples and register
+        if not sampled_seq._slm_mask.targets <= set(register.qubit_ids):
+            raise ValueError(
+                "The ids of qubits targeted in SLM mask"
+                " should be defined in register"
             )
-        self._tot_duration = self._seq.get_duration(
-            include_fall_time=self._modulated
-        )
-        self.samples_obj = sampler.sample(
-            self._seq,
-            modulation=self._modulated,
-            # The samples are extended by 1 to improve the ODE
-            # solver convergence
-            extended_duration=self._tot_duration + 1,
-        )
+        samples_list = []
+        for ch, ch_samples in sampled_seq.channel_samples.items():
+            if sampled_seq._ch_objs[ch].addressing == "Local":
+                # Check that targets of Local Channels are defined
+                # in register
+                if not set().union(
+                    *[slot.targets for slot in ch_samples.slots]
+                ) <= set(register.qubit_ids):
```

Comment:
Did you try this form?
```suggestion
                if not set().union(
                    slot.targets for slot in ch_samples.slots
                ) <= set(register.qubit_ids):
```

Reply:
I got (and get after re-try) `TypeError: unhashable type: 'set'`